### ISSUE TITLE
Inserting nested datatypes causes unexpected conversion of numerical type

### ISSUE THREAD

**Issue by n-hansen** [duckdb/duckdb#3955]
#### What happens?
If I create a table column with type `STRUCT(some_field DOUBLE)[]`, attempting to insert a row can trigger a SQLException complaining about a failed cast to `DECIMAL`.

#### To Reproduce
Executing the following two statements via the JDBC interface triggers a SQLException:
```sql
CREATE TABLE foo
 (my_struct STRUCT(my_double DOUBLE)[]);
```

```sql
INSERT INTO foo VALUES
  ([{'my_double': 1.33},
    {'my_double': 10.0}]);
```

```java.sql.SQLException: Conversion Error: Casting value "10.0" to type DECIMAL(3,2) failed: value is out of range!```

#### Environment (please complete the following information):
 - OS: OS X (intel)
 - DuckDB Version: 0.4.0
 - DuckDB Client: Java

#### Before Submitting

- [x] **Have you tried this on the latest `master` branch?**
Tested using [this master build](https://github.com/duckdb/duckdb/actions/runs/2555154383). 

- [x] **Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?**

This doesn't seem to be Java specific: I get the same behavior using the CLI client.

**Comment by Mytherin** [comment]
Thanks for the report! This is related to the system generating an incorrect combination of decimal types which causes the decimal cast to fail. The system uses the minimal decimal representation for all numerics in the parsing stage to avoid losing precision (so e.g. `10.0` becomes a `DECIMAL(3,1)`). This should be fixed in #3958.

**Comment by n-hansen** [comment]
Thanks for the quick turnaround!
